### PR TITLE
Partial instr review

### PR DESCRIPTION
**Pull request by nabobalis**
An incomplete review of the Python files for the instr submodule. 
I got like 80% of the way and I probably won't finish it but it seemed a waste not to PR what I did do.

### CODE REVIEW

**Comment by wtbarnes** [comment]
Thanks for doing this @nabobalis. I will do a review in the next day or so.

I think there needs to be some broader discussion about this whole subpackage. I did a very cursory look through it the other night and there's a lot of stuff in there, particularly in `goes.py`, that needs to be removed and spun out into affiliated packages. Maybe this could be a topic for an upcoming telecon...

**Comment by Cadair** [comment]
> that needs to be removed and spun out into affiliated packages

Agreed.

> Maybe this could be a topic for an upcoming telecon...

Sounds good.

**Comment by nabobalis** [comment]
Merge this and then you can discuss it.

**Comment by Cadair** [comment]
yes discussions about the future of the code are independent from this PR lol

**Comment by Cadair** [comment]
why didn't Azure run?

**Comment by nabobalis** [comment]
My token expired or hit a rate limit.

**Comment by nabobalis** [comment]
/azure-pipelines.yml: Could not get the latest source version for repository sunpy/azure-pipelines-template hosted on http://github.com/ using ref refs/heads/master. GitHub reported the error, "API rate limit exceeded for user ID 1392107."

File: sunpy/instr/tests/test_rhessi.py [hunk 1]
```diff
@@ -63,8 +61,8 @@ def test_parse_obssum_dbase_file():
 
 def test_parse_observing_summary_dbase_file():
     """
-    Test that we get the observing summary dbase file with the content
-    we expect.
+    Test that we get the observing summary dbase file with the content we
```
Comment (wtbarnes) [inline sunpy/instr/tests/test_rhessi.py]: dbase -> database? Unless I'm ignorant and "dbase" means something specific here...

File: sunpy/instr/tests/test_lyra.py [hunk 2]
```diff
@@ -48,7 +47,9 @@
 
 @pytest.mark.remote_data
 def test_split_series_using_lytaf():
-    '''test the downloading of the LYTAF file and subsequent queries'''
+    """
+    test the downloading of the LYTAF file and subsequent queries.
```
Comment (wtbarnes) [inline sunpy/instr/tests/test_lyra.py]: Capitalize "test"

File: sunpy/instr/tests/test_lyra.py [hunk 3]
```diff
@@ -97,7 +98,9 @@ def lyra_ts():
 
 @pytest.mark.remote_data
 def test_remove_lytaf_events_from_timeseries(lyra_ts):
-    """Test if artefacts are correctly removed from a TimeSeries."""
+    """
+    Test if artefacts are correctly removed from a TimeSeries.
```
Comment (wtbarnes) [inline sunpy/instr/tests/test_lyra.py]: "artefact" -> "artifact"? The latter seems to be the preferred American spelling, but I'm not sure what we've standardized on in the codebase as a whole...

File: sunpy/instr/lyra.py [hunk 4]
```diff
@@ -588,7 +591,9 @@ def get_lytaf_event_types(lytaf_path=None, print_event_types=True):
 
 
 def download_lytaf_database(lytaf_dir=''):
-    """download latest Proba2 pointing database from Proba2 Science Center"""
+    """
+    download latest Proba2 pointing database from Proba2 Science Center.
```
Comment (wtbarnes) [inline sunpy/instr/lyra.py]: Capitalize "download"

File: sunpy/instr/goes.py [hunk 5]
```diff
@@ -200,18 +200,17 @@ def calculate_temperature_em(goests, abundances="coronal",
...
     CHIANTI atomic physics database to model the response of the ratio
     of the short (0.5-4 angstrom) to long (1-8 angstrom) channels of the
-    XRSs onboard various GOES satellites.  This method assumes an
-    isothermal plasma, the ionisation equilibria of
+    XRSs onboard various GOES satellites. This method assumes an
+    isothermal plasma, the ionization equilibria of
     [2]_, and a constant density of 10**10 cm**-3.
     (See [1]_ for justification of this last assumption.)
-    This function is based on goes_chianti_tem.pro in SolarSoftWare
+    This function is based on "goes_chianti_tem.pro" in SolarSoft
```
Comment (wtbarnes) [inline sunpy/instr/goes.py]: "SolarSoftWare" is the full name. Do we consistently use "SolarSoft" elsewhere in the package?

File: sunpy/instr/goes.py [hunk 6]
```diff
@@ -152,42 +156,38 @@ def get_goes_event_list(timerange, goes_class_filter=None):
...
+    `~sunpy.timeseries.XRSTimeSeries`.
 
     This function calculates the isothermal temperature and
     corresponding volume emission measure of the solar soft X-ray
-    emitting plasma observed by the GOES/XRS.  This is done using the
+    emitting plasma observed by the GOES/XRS. This is done using the
     observed flux ratio of the short (0.5-4 angstrom) to long (1-8 angstrom)
-    channels.  The results are returned in a new LightCurve object which
-    contains metadata and flux data of the input LightCurve object in
+    channels. The results are returned in a new TimeSeries object which
```
Comment (wtbarnes) [inline sunpy/instr/goes.py]: Backticks around Timeseries so that it is linked to the API docs

File: sunpy/instr/fermi.py [hunk 7]
```diff
@@ -290,34 +330,36 @@ def nai_detector_angles():
...
+@add_common_docstring(**_variables_for_parse_time_docstring())
 def nai_detector_radecs(detectors, scx, scz, time):
     """
-    calculates the RA/DEC for each NaI detector given spacecraft z and x RA/DEC
-    positions.
+    calculates the "RA/DEC" for each NaI detector given spacecraft "z" and "x"
+    "RA/DEC" positions.
 
-    NB: This routine is based on code found in GTBURST, originally written by
+    This routine is based on code found in GTBURST, originally written by
```
Comment (wtbarnes) [inline sunpy/instr/fermi.py]: Should we provide some sort of reference or URL here?

Comment (nabobalis) [inline sunpy/instr/fermi.py]:
```suggestion
    This routine is based on code found in `GTBURST <https://fermi.gsfc.nasa.gov/ssc/data/analysis/scitools/gtburst.html>`__, originally written by
```

File: sunpy/instr/goes.py (hunk 6 above)
Comment (nabobalis) [inline sunpy/instr/goes.py]:
```suggestion
    channels. The results are returned in a new `~sunpy.timeseries.TimeSeries` object which
```

File: sunpy/instr/goes.py (hunk 5 above)
Comment (nabobalis) [inline sunpy/instr/goes.py]:
```suggestion
    This function is based on "goes_chianti_tem.pro" in SolarSoftWare
```

File: sunpy/instr/lyra.py (hunk 4 above)
Comment (nabobalis) [inline sunpy/instr/lyra.py]:
```suggestion
    Download latest Proba2 pointing database from Proba2 Science Center.
```

File: sunpy/instr/tests/test_lyra.py (hunk 2 above)
Comment (nabobalis) [inline sunpy/instr/tests/test_lyra.py]:
```suggestion
    Test the downloading of the LYTAF file and subsequent queries.
```

File: sunpy/instr/tests/test_lyra.py (hunk 3 above)
Comment (nabobalis) [inline sunpy/instr/tests/test_lyra.py]:
```suggestion
    Test if artifact are correctly removed from a TimeSeries.
```

File: sunpy/instr/tests/test_rhessi.py (hunk 1 above)
Comment (nabobalis) [inline sunpy/instr/tests/test_rhessi.py]:
```suggestion
    Test that we get the observing summary database file with the content we
```

**Comment by wtbarnes** [comment]
Can someone else give this a review? @DanRyanIrish?

File: sunpy/database/tests/test_attrs.py [hunk 8]
```diff
@@ -407,6 +407,13 @@ def test_walker_create_vso_instrument(vso_session):
     entries = walker.create(vso.attrs.Instrument('RHESSI'), vso_session)
     expected = [
         tables.DatabaseEntry(id=1, source=u'RHESSI', provider=u'LSSP',
+                             physobs=u'intensity',
```
Comment (Cadair) [inline sunpy/database/tests/test_attrs.py]: this is causing a test to fail.

Comment (wtbarnes) [reply sunpy/database/tests/test_attrs.py]: It looks like the second and third entries in `expected` are the same? Should the last one be removed? Or should we actually expect 3 entries from `vso_session` for RHESSI?

Comment (nabobalis) [reply sunpy/database/tests/test_attrs.py]: I'll fix this tonight

**Comment by nabobalis** [comment]
Online test failure seems unrelated.

**Comment by wtbarnes** [comment]
🎉 thanks @nabobalis

**Pull request merged**